addons/mrp/migrations/12.0.2.0/post-migration.py

```diff
@@ -36,6 +36,23 @@ def fill_stock_warehouse_picking_types(env):
             warehouse._create_or_update_sequences_and_picking_types())
 
 
+def force_set_products_consumable(env):
+    """Set products are only allowed to be consumable
+    as per https://github.com/odoo/odoo/commit/3d34d5838"""
+    openupgrade.logged_query(
+        env.cr,
+        """
+        UPDATE product_template SET type = 'consu'
+        WHERE id IN (
+            SELECT pt.id FROM product_product pp
+            JOIN product_template pt ON pt.id = pp.product_tmpl_id
+            WHERE pt.type = 'product' AND EXISTS(
+                SELECT * FROM mrp_bom mb
+                WHERE mb.type = 'phantom' AND mb.active
+                    AND (mb.product_tmpl_id = pt.id OR pp.id = mb.product_id)))
+        """)
```

Comment:
For only writing those needed (it can be also an indicator of what was wrong configured):
```suggestion
        AND type != 'consu'""")
```

Reply:
I'm reverting your suggestion because I already had pt.type = 'product' in the sub clause.

Reply:
Is it possible to have previously the product as service?

Reply:
If anybody configures their set products as services they probably have a good reason to do so, and in that case they may override the [method that enforces the product type](https://github.com/odoo/odoo/commit/3d34d5838#diff-cadd7a2530955f21c93bbe4372640d0bbb4afce512c9a2b0b6f0dbbc96585096R83), I imagine ;-)